packages/passport/README.md

```diff
@@ -0,0 +1,32 @@
+# Nest Middlewares - Passport
+
+[![Coverage Status](https://coveralls.io/repos/github/wbhob/nest-middlewares/badge.svg?branch=master)](https://coveralls.io/github/wbhob/nest-middlewares?branch=master)
+
+This is the Nest Middleware wrapper around [passport](http://www.npmjs.com/package/passport).
+
+## Installation
+
+`@nest-middlewares/passport` is available from NPM. You can install it with this command:
+
+```sh
+npm install --save @nest-middlewares/passport
+```
+
+## Usage
+
+```ts
+import { Module, NestModule, MiddlewareConsumer } from '@nestjs/common';
+import { PassportInitializeMiddleware, PassportSessionMiddleware } from '@nest-middlewares/passport';
+import { AppController } from './app.controller';
+
+@Module(...)
+export class AppModule implements NestModule {
+    configure(consumer: MiddlewaresConsumer) {
+        consumer
+        	.apply(PassportInitializeMiddleware)
```

Comment:
I don't know if this config will work (I think you need to reference consumer twice, as forRoutes doesn't apply anything. Also, forRoutes takes an object as far as I know).

Reply:
I'm using it here [https://gitlab.com/revant.one/identity-provider] and it is working, i.e Session getting created and stored into Typeorm session store, Browser shows logged in user, `req.user` persists until logout.

I faced another problem with this setup, I can't run `test:e2e`, it fails with

```
$ yarn test:e2e
yarn run v1.7.0
$ jest --config ./test/jest-e2e.json --forceExit
 FAIL  test/app.e2e-spec.ts
  AppController (e2e)
    ✕ /GET / (25ms)

  ● AppController (e2e) › /GET /

    TypeError: Cannot read property 'initialize' of undefined
      
      at Object.<anonymous>.Authenticator.initialize (../node_modules/passport/lib/authenticator.js:133:26)
      at PassportInitializeMiddleware.resolve (../src/middlewares/passport.middlewares.ts:6:29)
      at MiddlewareModule.bindHandler (../node_modules/@nestjs/core/middleware/middleware-module.js:80:34)
      at Promise.all.middlewareCollection.map (../node_modules/@nestjs/core/middleware/middleware-module.js:68:24)
          at Array.map (<anonymous>)
      at MiddlewareModule.registerRouteMiddleware (../node_modules/@nestjs/core/middleware/middleware-module.js:61:48)
      at Promise.all.forRoutes.map (../node_modules/@nestjs/core/middleware/middleware-module.js:56:24)
          at Array.map (<anonymous>)
      at MiddlewareModule.registerMiddlewareConfig (../node_modules/@nestjs/core/middleware/middleware-module.js:55:37)
      at middlewareConfig.map (../node_modules/@nestjs/core/middleware/middleware-module.js:47:24)
          at Array.map (<anonymous>)
      at registerAllConfigs (../node_modules/@nestjs/core/middleware/middleware-module.js:46:83)
      at Promise.all.map (../node_modules/@nestjs/core/middleware/middleware-module.js:50:31)
          at Array.map (<anonymous>)
      at MiddlewareModule.registerMiddleware (../node_modules/@nestjs/core/middleware/middleware-module.js:49:50)
      at NestApplication.registerMiddleware (../node_modules/@nestjs/core/nest-application.js:251:37)
      at NestApplication.registerRouter (../node_modules/@nestjs/core/nest-application.js:117:20)
      at NestApplication.init (../node_modules/@nestjs/core/nest-application.js:88:20)
          at <anonymous>
      at process._tickCallback (../internal/process/next_tick.js:188:7)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        3.563s
Ran all test suites.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
Anything I missed?

Reply:
solved it,

`process.env.NODE_ENV==='test'` behaves differently.

https://gitlab.com/revant.one/identity-provider/blob/fe83b64aa56ae08e59cbda7cd55adfc749b14a50/src/auth/auth.module.ts#L25

Reply:
did you make the changes I requested above?

Reply:
> did you make the changes I requested above?

No, I'll get back.
Trying to understand things.

> Also, you'll need to update the lockfile for the CI to turn green (rm -r node_modules; rm package-lock.json; npm install).

package-lock.json I understood. 

> if you could also add some tests to make sure everything is initializing correctly, that would be a big help.

I'll try this. I can't promise anything.

> If you don't know how to do this, I can give you a hand with it. :) cheers!

That will be great!